testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -660,6 +661,42 @@ else if (!(isParquetStorageFormat && isParquetOptimizedWriterEnabled)) {
         onTrino().executeQuery(format("DROP TABLE %s", tableName));
     }
 
+    @Test(groups = STORAGE_FORMATS_DETAILED)
+    public void testTimestampFieldWrittenByOptimizedParquetWriterCannotBeReadByHive()
```

Comment:
Could be worth having two tests, one with a short timestamp precision and one with a long timestamp precision.